app/ui/src/main/java/com/fsck/k9/activity/MessageCompose.java

```diff
@@ -753,7 +756,7 @@ private void checkToSaveDraftImplicitly() {
             return;
         }
 
-        if (!changesMadeSinceLastSave) {
+        if (!changesMadeSinceLastSave || !draftIsNotEmpty()) {
```

Comment:
I thought this check in "checkToSaveDraftImplicitly" would probably just be consistent as we check for changes and emptyness eg. on "prepareForFinish" too. https://github.com/k9mail/k-9/blob/928400a2338d21b5605556988cfe96ee2573fa0a/app/ui/src/main/java/com/fsck/k9/activity/MessageCompose.java#L1049

Reply:
@cketti I'm trying to rewrite these things now. First, avoiding this draftIsNotEmpty check there is not that easy. When we open a new message the system sets these always bcc addresses, so changesMadeSinceLastSave is already true without having clicked anywhere (and having an empty draft though). Second, trying to compare all bcc recipients with alwaysBcc recipients in draftIsNotEmpty is tricky too. Because on an account change, the alwaysBcc array is updated directly but the bcc view is going through an async recipient loader first. So even if the values should be equal on a check, they always mismatch.

Reply:
Ideally you'd modify the "auto BCC" mechanism so it doesn't modify the value of `changesMadeSinceLastSave`.